src/views/charts/PnlChart.tsx

```diff
@@ -168,9 +141,37 @@ export const PnlChart = ({
                 } as PnlDatum)
             )
         : [],
-    [pnlData, equity, selectedPeriod, now]
+    [pnlData, equity?.current, now]
   );
 
+  const getPeriodOptions = (oldestPnlMs: number): HistoricalPnlPeriods[] => {
+    const availablePeriods: HistoricalPnlPeriods[] = [HistoricalPnlPeriod.Period1d];
+
+    _.forEach(Object.entries(MS_FOR_PERIOD), ([, ms], i, entries) => {
+      if (oldestPnlMs < now - ms) {
+        const nextPeriod = _.get(entries, [i + 1, 0]);
+        if (nextPeriod)
+          availablePeriods.push(
+            HISTORICAL_PNL_PERIODS[nextPeriod as keyof typeof HISTORICAL_PNL_PERIODS]
+          );
+      }
+    });
+
+    return availablePeriods;
+  };
+
+  useEffect(() => {
+    const oldestPnlMs = pnlData?.[0]?.createdAtMilliseconds;
```

Comment:
[q] is there a reason we can't define this outside the useEffect and use the var in the dependency array? it will be very slightly nicer to keep this synced w/ the dep array (there has been a case in order components where values get out of sync and functions behave strangely)